### ISSUE TITLE
Why does android own Selinux affect p-root?

### ISSUE THREAD

**Issue by Root-voidX** [corbinlc/GNURootDebian#162]
To anyone who has any idea. When trying to execute a package to visualize data selinux stops it from launching. Selinux stops anything from running as an executable stack. So is there away to stop the device selinux from accessing the container ? From corbinlc last github he said an he did some changes on this but ever since mm selinux has been a pain in my butt. Is there away to make exceptions?

**Comment by corbinlc** [comment]
Can you provide a small example of how to start from a fresh GNURoot Debian install and end up with this error?  A preloaded library we use called libdisableselinux.so has historically taken care of most selinux issues, but it may need to be improved in some way.

**Comment by Root-voidX** [comment]
Well these are the steps from a fresh install:
-install gnuroot debian
-install xterm
-run "apt-get update & apt-get upgrade"
-then install either paraview or blender 
-when launching blender or paraview from xterm bash will spit out an error for libx264.142.so. It cannot run as shared executable stack.

**Comment by Root-voidX** [comment]
Yeah it does seen like a selinux issue at best. MM on my s7e didn't have selinux or had exceptions but once upgraded to nougat which enforce its, it stopped. It doesn't work on any nexus of lg device either. Glxgears works so opengl rendering works, I have full access to the soc, just can't do kernal stuff.

**Comment by vandijk72** [comment]
Sorry if this is slightly off-topic (I am using XSDL rather than native X11) but as the distribution's version of paraview gave me the same error, I tried compiling Paraview 5.3 from source using a GnuRootDebian installation on a tab S3 (this takes a few days with make -j 2). I couldn't get it to run until I installed libgl1-mesa-swx11-dev.  Then paraview helpfully pointed out that my GL library does not support the GLX 2 extensions version 5 needs, so I went back to Paraview 4.4, which works, It's slow as I suppose software rendering is being used.  Below is the cmake command I used:

cmake \
-DCMAKE_BUILD_TYPE=Release \
-DBUILD_TESTING=OFF \
-DBUILD_SHARED_LIBS=ON \
-DPARAVIEW_USE_MPI=OFF \
-DPARAVIEW_ENABLE_PYTHON=OFF \
../ParaView-v4.4.0-source

I imagine other flags would work as well.  To make interaction a bit better I set paraview to outline for 0 cells in interactive mode.  Do you think it might be possible to use some hardware rendering somehow?

Anyway, slow visualisation is immeasurably  better than no visualisation, so I am already happy.  Perhaps X2Go, which seems to work straight out of the distribution, is another option if you have an internet connection to a server.  Basically, GnuRootDebian is amazing. Thanks Corbin!

**Comment by Root-voidX** [comment]
I have used xsdl for many linux apps but yes its software rendering. You could speed it up by using a more lighter DE. But back in MM on my samsung device, paraview and blender from the debian repo worked fine. It was only in nougat that it stopped when selinux was introduced. I never got glxgears running on xsdl but on x11 it did. There should be a way to configure x11 to directly boot into a DE of your choice to. You could also use openmpi to accelerate visualization using a pc too! 

Did you get blender to work on xsdl? 

I have seen some crazy multi physics simulation using cfd and other codes and it would be sweet to see it on arm!!

I still dont know if selinux policy was changed on nougat in such away that it is causing this!

**Comment by vandijk72** [comment]
I haven't used blender before, so I have not tried to install it. I just need paraview to help those using my own software.  

I tried to get the native x11 client (which provides twm?) to display something on my phone (running MM), but it doesn't authorise my opening of the display.  I will try again when I have more time.  Anyway my tablet is running nougat so I guess it's xsdl/swx11 for me for the next while (glxgears/info works with this combo). I use LXDE, which at least seems pretty fast. 

Like you say, running on arm is nice. For my own CFD code, the tablet (snapdragon 820) runs at 1/3 to 1/2 the speed of my Xeon desktop, which is excellent considering its relative size and weight.

**Comment by Root-voidX** [comment]
You would have to launch xterm and then use bash from within to launch apps on xterm! I haven't found a default configuration to launch straight from terminal. Also I double checked on a nexus device with selinux disabled and it runs great!! Also yes glxgears and glxsphere run better in xterm, guess its hardware accelerated

**Comment by Root-voidX** [comment]
Also there seems to still be an open issue on xsdl github about glxgears running poorly or not at all so right now xterm seems supreme, its just selinux policies..

**Comment by Root-voidX** [comment]
I also notice its impossible to use backspace in any apps in xterm as well. Double checking with my note 3 on 5.1 without selinux and it works fine. My s7e back on 6.1 had no issue either. I believe whichever method used, current selinux is to restricted for 6.0+

**Comment by Root-voidX** [comment]
https://imgur.com/BYUFDlI

With selinux set to permissive I can run this otherwise it won't work. And to do that I need root access? Maybe anything above 6.0 won't play well due to selinux :/

**Comment by Root-voidX** [comment]
Has anyone figured out how I can ignore selinux without root? As you saw from my last comment, I needed to be rooted to change selinux. Really hampers usage on newer devices.

With selinux set to permissive I can do this: https://imgur.com/BYUFDlI

But I used to be able to do this on samsung devices with MM but with nougat and selinux it wont allow....

**Comment by Root-voidX** [comment]
@corbinlc any updates on this ?

**Comment by the-maux** [comment]
@Root-voidX SeLinux is  a kernel module, so to change the set you have to be root, no user will be able to modify kernel stuff, ever, or its a privilege escalation and its a an exploit.
Imo, your only solution is to flash directly your device with a kernel set to permissive.
If you don't find it on the web, you can still recompile your kernel with the new permission, so it will always be on permissive.

**Comment by Root-voidX** [comment]
@the-maux I see. Seems like oem images have been becoming more restrictive. I have a nexus 7 which is rooted and I'm able to use root access on terminal emulator to set SELinux to permissive and it works. So I would need root access to use the payload/packages because of SELinux. Kind of sucks that I have to be rooted since I use this app because it didn't require root.

**Comment by Root-voidX** [comment]
@corbinlc or anyone else who helps maintain this app, is there any upcoming beta release that updates libdisableselinux.so or fixes the selinux issue with p-root and or updating to stretch? I had no issues with 5.0 or lower, only 6.0 or higher. Most of the new release have changes that greatly help performance. Would be nice to run them without root.

Cheers!

**Comment by AdamMatthewLTM** [comment]
Hi all,

I am currently assisting the developers in sorting relevant issues for GNURoot Debian on GitHub. If the issue discussed is still relevant, could you please reply to this thread and state the issue needs to be resolved and I will attempt to sort it for you ASAP. If the issue is no longer relevant, please reply to this thread and state it has been resolved, and I will suggest this thread be closed.

Furthermore, if a particular issue is considered to be high priority, please state that you are requesting it to be considered as high priority and I will mark it as such.

Thank you all so much for your time and understanding,
-Adam

**Comment by Root-voidX** [comment]
Yes can you please find a way to go around android   8 and possibly android 9 SElinux policies?
